test/DefaultCluster.Tests/GrainReferenceTest.cs

```diff
@@ -248,7 +248,7 @@ private T NewtonsoftJsonSerializeRoundtrip<T>(T obj)
             var settings = OrleansJsonSerializer.GetDefaultSerializerSettings(this.HostedCluster.SerializationManager, this.GrainFactory);
             // http://james.newtonking.com/json/help/index.html?topic=html/T_Newtonsoft_Json_JsonConvert.htm
             string json = JsonConvert.SerializeObject(obj, settings);
-            object other = JsonConvert.DeserializeObject(json, settings);
+            object other = JsonConvert.DeserializeObject(json, typeof(T), settings);
```

Comment:
could this *fix* be an indicator that we broke some back compat (whether implicitly or explicitly)?